linux_os/guide/system/logging/rsyslog_filecreatemode/oval/shared.xml

```diff
@@ -0,0 +1,112 @@
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("rsyslog will create logfiles that do not already exist on the system. This setting controls what permissions will be applied to these newly created files.") }}}
+    <criteria operator="ONE">
```

Comment:
It was indeed chosen to be `ONE`, because it is not about the value of $FileCreateMode, but if it is defined in both places.
Let's say you have `$FileCreateMode 0600` in `/etc/rsyslog.conf` and you have `$FileCreateMode 0640` in `/etc/rsyslog.d/99-rsyslog_filecreatemode.conf`, both values are valid and will lead to both criterias being `True`. Then the `ONE` will make it fail as only one of the criterias should be `True`.

If you have `OR` in the case above, then it will pass when it shouldn't.

Therefore I believe the current check already covers your scenario.

Reply:
If we try to have the check you mentioned first, and then a variable for checking the value, I still think you will need two variables to convert from octal. Because you will have two tests, consequently two objects, to check `rsyslog.conf` and `rsyslog.d`. You will need a variable for each object, right?
Or do you have something else in mind?

Reply:
Thanks for the clarification @dodys . I forgot about the ONE value and didn't find another use in the project. But checked the [OVAL documentation](https://oval-community-guidelines.readthedocs.io/en/latest/oval-schema-documentation/oval-common-schema.html?#operatorenumeration-1) now and you are correct.

Reply:
> If we try to have the check you mentioned first, and then a variable for checking the value, I still think you will need two variables to convert from octal. Because you will have two tests, consequently two objects, to check `rsyslog.conf` and `rsyslog.d`. You will need a variable for each object, right? Or do you have something else in mind?

My point was more about the duplication of code and I was wondering if it could be somehow improved. Maybe a new OVAL macro would interesting since this conversion is used in at least 5 other rules. It would be great, but I would be fine leaving this out of scope in this PR. Maybe a separate PR just for the macro and the update of the impacted rules would be better.

Besides the macro, the other idea would demand to change the approach. For example:
First Test:
- Collect the FileCreateMode value in both `/etc/rsyslog.conf` and `rsyslog.d`.
  - It would still demand two objects, like now.
- Concatenate the objects in a new variable
  - It would demand only one conversion. We could consider only the first object, for example, in case the concatenated variable contains multiple objects.
- Compare the variable object to the expected state.

Second Test
- Ensure that concatenated variable contains only one object.

It would reduce the number of tests from 4 to 2 and would reduce code duplication.
Maybe it would be worth to consider.

Reply:
I think I got it now, let me know what you think